docs/modules/ROOT/pages/api/upgrades.adoc

```diff
@@ -0,0 +1,86 @@
+:github-icon: pass:[<svg class="icon"><use href="#github-icon"/></svg>]
+:class_hash: https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/class-hash/[class hash]
+
+= Upgrades
+
+Reference of interfaces and utilities related to upgradeability.
+
+== Core
+
+[.contract]
+[[IUpgradeable]]
+=== `++IUpgradeable++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/cairo-2/src/upgrades/interface.cairo#L3[{github-icon},role=heading-link]
+
+:Upgraded: xref:Upgradeable-Upgraded[Upgraded]
+
+```javascript
+use openzeppelin::upgrades::interface::IUpgradeable;
+```
+
+Interface of an upgradeable contract.
+
+[.contract-index]
+.Functions
+--
+* xref:#IUpgradeable-upgrade[`++upgrade(new_class_hash)++`]
+--
+
+[#IUpgradeable-Functions]
+==== Functions
+
+[.contract-item]
+[[IUpgradeable-upgrade]]
+==== `[.contract-item-name]#++upgrade++#++(new_class_hash: ClassHash)++` [.item-kind]#external#
+
+Upgrades the contract by updating the contract {class_hash}.
+
+NOTE: This function is usually protected by an xref:access.adoc[Access Control] mechanism.
```

Comment:
Should we rephrase this as: `This function should be protected...`?

Reply:
mmm, The only issue I have with that, is that users can protect the function using a different way right? Like timestamps, or I don't know, but doesn't have to be AccessControl (even when usually it is).

Reply:
Right! Good point